modules/nw-ingress-route-secret-load-external-cert.adoc

```diff
@@ -0,0 +1,76 @@
+// Module included in the following assemblies:
+//
+// * ingress/routes.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="nw-ingress-route-secret-load-external-cert_{context}"]
+= Creating route with externally managed certificate
+
+:FeatureName: Securing route with external certificates in TLS secrets
+include::snippets/technology-preview.adoc[]
+
+You can configure OpenShift routes with third-party certificate management solutions by using the `.spec.tls.externalCertificate` field of the route API. You can reference externally managed TLS certificates via secrets, eliminating the need for manual certificate management. Using the externally managed certificate reduces errors and provides faster updates. 
+
+[NOTE]
+====
+This feature applies to both edge routes and re-encrypt routes.
+====
+
+.Prerequisites
+
+* You must enable the `RouteExternalCertificate` feature gate.
+* You must have the `create` and `update` permissions on the `routes/custom-host` sub-resource.
+* You must have a secret containing a valid certificate/key pair in PEM-encoded format of type `kubernetes.io/tls`, which includes both `tls.key` and `tls.crt` keys.
+* You must place the referenced secret in the same namespace as the route you want to secure.
+
+.Procedure
+
+. Create a `role` in the same namespace as the secret to allow the router service account read access.
++
+[source,terminal]
+----
+$ oc create role secret-reader --verb=get,list,watch --resource=secrets --resource-name=<secret-name> \ <1> 
+--namespace=<current-namespace> <2>
+----
+<1> Replace <secret-name> with the actual name of your secret.
+<2> Replace <current-namespace> with the namespace where both your secret and route reside.
+
+. Create a `rolebinding` in the same namespace as the secret and bind the router service account to the newly created role.
++
+[source,terminal]
+----
+$ oc create rolebinding secret-reader-binding --role=secret-reader --serviceaccount=openshift-ingress:router --namespace=<current-namespace> <1>
+----
+<1> Replace <current-namespace> with the namespace where both your secret and route reside.
+
+. Create a `route` specifying the `.spec.tls.externalCertificate` field to reference the secret containing your certificate:
++
+.YAML Definition of the Secure Route
+[source,yaml]
+----
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: myedge
+  namespace: test
+spec:
+  host: myedge-test.apps.example.com
+  tls:
+    externalCertificate:
+      name: <secret-name> <1>
+    termination: edge
+    [...]
+[...]
+----
+<1> Replace <secret-name> with the actual name of your secret.
+
+If the secret exists and has a certificate/key pair, the router will serve the generated certificate if all prerequisites are met.
+
+[NOTE]
+====
+If `.spec.tls.externalCertificate` is not provided, the router will use default generated certificates. You cannot provide both `.spec.tls.certificate`/`.spec.tls.key` and `.spec.tls.externalCertificate`.
```

Comment:
```suggestion
If `.spec.tls.externalCertificate` is not provided, the router will use default generated certificates. You cannot provide both the `.spec.tls.certificate` field  and the `.spec.tls.key` field if you are providing the `.spec.tls.externalCertificate` field.
```

Reply:
Or you could rephrase to: You can only provide the `.spec.tls.certificate` field or the `.spec.tls.key` field if you are using the `.spec.tls.externalCertificate` field in your route configuration.